store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/operation/handler/utilities/WriteUnsortedData.java

```diff
@@ -84,17 +84,17 @@ private void _writeElements(final Iterator<? extends Element> elements) throws O
         while (elements.hasNext()) {
             final Element element = elements.next();
             final String group = element.getGroup();
-            final ParquetWriter<Element> writer;
+            ParquetWriter<Element> writer = null;
             final Map<Integer, ParquetWriter<Element>> splitToWriter;
-            if  (groupSplitToWriter.containsKey(group)) {
+            if (groupSplitToWriter.containsKey(group)) {
                 splitToWriter = groupSplitToWriter.get(group);
             } else {
                 splitToWriter = new HashMap<>();
                 groupSplitToWriter.put(group, splitToWriter);
             }
             if (schemaUtils.getEntityGroups().contains(group)) {
                 writer = getWriter(splitToWriter, groupToSplitPoints.get(group), ((Entity) element).getVertex(), group, ParquetStoreConstants.VERTEX);
-            } else {
+            } else if (schemaUtils.getEdgeGroups().contains(group)) {
                 writer = getWriter(splitToWriter, groupToSplitPoints.get(group), ((Edge) element).getSource(), group, ParquetStoreConstants.SOURCE);
             }
```

Comment:
Can we have a final else to log the fact that the element is ignored if it's not in schema - probably DEBUG level logging?

Reply:
This is on the four lines below, but at a WARN level.

Reply:
So it is, ignore my comment.